src/tencent/config.rs

```diff
@@ -0,0 +1,95 @@
+use std::collections::HashMap;
+use std::env;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+use std::sync::RwLock;
+
+use super::constants::*;
+
+#[derive(Clone, Default)]
+#[cfg_attr(test, derive(Debug))]
+struct Config {
+    /// `access_key_id` will be loaded from
+    ///
+    /// - this field if it's `is_some`
+    /// - env value: [`TENCENT_CLOUD_ACCESS_KEY_SECRET`]
+    secret_key: Option<String>,
+    /// `access_key_secret` will be loaded from
+    ///
+    /// - this field if it's `is_some`
+    /// - env value: [`TENCENT_CLOUD_ACCESS_KEY_ID`]
+    secret_id: Option<String>,
+
+    /// `security_token` will be loaded from
+    ///
+    /// - this field if it's `is_some`
+    security_token: Option<String>,
+}
+
+/// Config loader that will load config from different source.
+#[derive(Default)]
+#[cfg_attr(test, derive(Debug))]
+pub struct ConfigLoader {
+    config: Arc<RwLock<Config>>,
+
+    /// Mark whether we have read env or not.
+    read_env: AtomicBool,
+}
+
+impl Clone for ConfigLoader {
+    fn clone(&self) -> Self {
+        Self {
+            config: self.config.clone(),
+            read_env: AtomicBool::from(self.read_env.load(Ordering::Relaxed)),
+        }
+    }
+}
+
+impl ConfigLoader {
+    pub fn load_via_env(&self) {
+        if self.read_env.load(Ordering::Relaxed) {
+            return;
+        }
+        self.read_env.store(true, Ordering::Relaxed);
+
+        let envs = env::vars().collect::<HashMap<_, _>>();
+        let mut config = { self.config.read().expect("lock must be valid").clone() };
+
+        if let Some(v) = envs.get(TENCENT_CLOUD_ACCESS_KEY_SECRET) {
```

Comment:
**question**: Do you have references that tencent cloud will use this env? If no SDKs adopt this, we should not do this either.

Reply:
OK,Previously I thought it was required to be defined by the project